### PR TITLE
feat: twitch client

### PR DESCRIPTION
**Pull request by lino-levan**
towards #42

### CODE REVIEW

File: src/create_oauth2_client.ts [hunk 1]
```diff
@@ -169,6 +169,41 @@ export function createSlackOAuth2Client(
...
+ * 1. TWITCH_CLIENT_ID
+ * 2. TWITCH_CLIENT_SECRET
+ *
+ * ```ts
+ * import { createTwitchOAuth2Client } from "https://deno.land/x/deno_kv_oauth/mod.ts";
+ *
+ * const oauth2Client = createTwitchOAuth2Client({
+ *  redirectUri: "http://localhost:8000/callback",
+ *  defaults: {
+ *    scope: "users.read"
```
Comment (iuioiua) [inline src/create_oauth2_client.ts:185]: Is this scope valid for this given OAuth2 client?

File: src/create_oauth2_client.ts [hunk 2]
```diff
@@ -169,6 +169,41 @@ export function createSlackOAuth2Client(
...
+ *  }
+ * });
+ * ```
+ *
+ * @see {@link https://dev.twitch.tv/docs/authentication/getting-tokens-oauth/#authorization-code-grant-flow}
+ */
+export function createTwitchOAuth2Client(
+  additionalOAuth2ClientConfig?:
```
Comment (iuioiua) [inline src/create_oauth2_client.ts:193]:
```suggestion
  additionalOAuth2ClientConfig:
```
We should do this for any other functions requiring redirect URI and/or scope in another PR.

**Review by iuioiua — CHANGES_REQUESTED** [review]
I'm getting an HTTP 400 error after setting the correct scope. Is this working for you?

File: src/create_oauth2_client.ts (hunk 1 above)
Comment (lino-levan) [reply src/create_oauth2_client.ts:185]: Oh shoot, I forgot to change that. Apologies.

File: src/create_oauth2_client.ts (hunk 2 above)
Comment (lino-levan) [reply src/create_oauth2_client.ts:193]: Yup, makes sense. I'll fix twitch's in this PR and open another one fixing all of them once this gets merged.

Comment (iuioiua) [reply src/create_oauth2_client.ts:193]: Cool. I've already done this for the others in #80.

Comment (lino-levan) [reply src/create_oauth2_client.ts]: Oh nice, sounds good.

**Comment by lino-levan** [comment]
Twitch doesn't support PKCE. This is going to be more complex of a PR than I thought. Will give myself a night's rest to try to come up with a clean solution to this.

**Comment by lino-levan** [comment]
The joys of creating an OAuth2 library.

**Comment by lino-levan** [comment]
Had a chat with @iuioiua online. No PKCE is a dealbreaker. Closing this PR. Please make some noise if you need this anyways!

**Pull request closed without merging**